ujt/transformers.py

```diff
@@ -0,0 +1,65 @@
+""" Module holding transformer functions.
+
+Transformer functions take a Dash-specific data structure or an intermediate data structure,
+make a change, and return the same type of data structure.
+"""
+
+from collections import deque
+
+from . import constants, state
+
+
+def apply_highlighted_edge_class_to_elements(elements, user_journey_name):
+    edges_map = {}
+    nodes_list = []
+
+    for element in elements:
+        element_data = element["data"]
+        if "source" in element_data.keys():
+            edges_map[(element_data["source"],
+                       element_data["target"])] = element
+        else:
+            nodes_list.append(element)
+
+    updated_edges_map = remove_highlighted_class_from_edges(edges_map)
+
+    if user_journey_name:
+        updated_edges_map = apply_highlighted_class_to_edges(
+            updated_edges_map,
+            user_journey_name)
+
+    return list(updated_edges_map.values()) + nodes_list
+
+
+def remove_highlighted_class_from_edges(edges_map):
+    for edge in edges_map.values():
+        edge["classes"] = ""
+    return edges_map
+
+
+def apply_highlighted_class_to_edges(edges_map, user_journey_name):
```

Comment:
We use a BFS traversal to apply the a highlighted class to the edges so they stylesheet can color them differently.
Chose to use iteration over recursion here as it's easier to test. 

Note that here we traverse the actual protos (node_name_message_map). A future branch refactors this to only traverse the Cytoscape elements (to support virtual nodes)